File: program/program/convert-binary-to-decimal/ConvertBinaryToDecimal.js
/** 
* @param {number} binaryNumber - the binary number to convert, in number form (not string)
* @returns {number} the converted number in decimal 
*/

function convertBinaryToDecimal(binaryNumber){
    const digitsArray = binaryNumber.toString().split('')

    const numberOfDigits = digitsArray.length

    const multiplicatedDigits = digitsArray.map((digit, index) => {
        let power = numberOfDigits - index - 1
        return parseInt(digit) * Math.pow(2, power)
    })

    const resultSum = multiplicatedDigits.reduce((total, accumulator) => {
        return total + accumulator
    }, 0)

    return resultSum
}

/** 
// Test Cases
console.log(convertBinaryToDecimal(1010))
// should return 10
console.log(convertBinaryToDecimal(1111))
// should return 15
*/

File: program/program/convert-binary-to-decimal/convert_binary_to_decimal.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var num, rem int
	fmt.Scanf("%d", &num)
	index := 0
	dnum := 0
	for num != 0 {
		rem = num % 10
		num = num / 10
		dnum = dnum + rem*int(math.Pow(2, float64(index)))
		index++
	}
	fmt.Println(dnum)
}
